### ISSUE TITLE
Sequentiality of NRTM serials

### ISSUE THREAD

**Issue by mxsasha** [irrdnet/irrd4#85]
It was my understanding that NRTM serials are always sequential - unfortunately it seems this is not the case for ARIN:
```
$ telnet rr.arin.net 4444
Trying 199.71.0.110...
Connected to rr.arin.net.
Escape character is '^]'.
-g ARIN:3:96380-LAST

% The ARIN Database is subject to Terms and Conditions.
% See http://www.arin.net/db/support/db-terms-conditions.pdf

%START Version: 3 ARIN 96380-96381 FILTERED

ADD 96381

mntner:         MNT-AS268354
descr:          Dummy description for MNT-AS268354
admin-c:        DUMY-ARIN
upd-to:         unread@arin.net
auth:           MD5-PW $1$anntMjx4$h5F7JUovDWX6XVwrmBeN8/
mnt-by:         MNT-AS268354
changed:        unread@arin.net 20000101
source:         ARIN

%END ARIN
```

Serial 96380 does not seem to exist. Requesting this serial only results in a sort of empty response:
```
$ telnet rr.arin.net 4444
Trying 199.71.0.110...
Connected to rr.arin.net.
Escape character is '^]'.
-g ARIN:3:96380-96380

% The ARIN Database is subject to Terms and Conditions.
% See http://www.arin.net/db/support/db-terms-conditions.pdf

%START Version: 3 ARIN 96380-96380 FILTERED

%END ARIN
```

Is this perhaps an update to a person object, or some other object that was filtered out? A bug in ARIN's database? A protocol feature? Currently, we reject this because per serial that falls in the START range, we expect one operation, but we can remove this validation.

(Currently, IRRDv4 guarantees gapless serials for updates to the authoritative DB, but dropping this requirement, causing gaps in less common cases, may improve update processing performance a bit.)

**Comment by mxsasha** [comment]
This issue would also be worse for NRTMv1 streams like LEVEL3, because there the only way to know the serial of an operation is by counting based on the serials noted in the START header - and if there are any gaps, we can't know which serial was skipped.

One solution could be to allow gaps in serials, but only in NRTMv3 streams - which includes ARIN.

**Comment by job** [comment]
I agree with your proposed solution. Allow gaps in v3 (but only increases,
we won’t allow going down)
